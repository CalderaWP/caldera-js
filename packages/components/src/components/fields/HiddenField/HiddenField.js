import {fieldClassNames, FieldWrapper} from "../../..";
import React from "react";
import {parseAttributes} from "../util";

export const HiddenField = ({value,fieldId,onChange,attributes}) => {
    const fieldType = 'hidden';
    const _attributes = parseAttributes(attributes, fieldType);


    return (
        <FieldWrapper fieldType={fieldType}>
            <input
                className={fieldClassNames(fieldType)}
                type={'hidden'}
                id={fieldId}
                value={value}
                onChange={(event) => onChange(event.target.value)}
                {..._attributes}
            />
        </FieldWrapper>
    )
}