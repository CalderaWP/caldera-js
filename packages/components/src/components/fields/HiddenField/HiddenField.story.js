import React from 'react';
import { storiesOf } from '@storybook/react';
import { HiddenField } from './HiddenField';

const onChange = newValue => {
	console.log(newValue);
};

storiesOf('Hidden Field', module).add('You can not see this field', () => (
	<HiddenField fieldId={'i11'} value={'Roy'} onChange={onChange} />
));
