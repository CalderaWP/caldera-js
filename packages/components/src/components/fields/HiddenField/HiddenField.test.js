import * as React from 'react';
import {render} from 'react-testing-library';
import { mount } from 'enzyme';

import { HiddenField } from './HiddenField';

describe('HiddenField ', () => {
	let onChange;

	beforeEach(() => {
		onChange = jest.fn();
	});

	it('matches snapshot with all props', () => {
		const component = render(
			<HiddenField
				fieldId={'i11'}
				value={'Roy'}
				onChange={onChange}
			/>
		);
		expect(component).toMatchSnapshot();
	});


	it('Changes calls change handler', () => {
		const component = mount(
			<HiddenField
				fieldId={'i11'}
				value={'Roy'}
				onChange={onChange}
			/>
		);

		expect( component.find('input').prop('value')).toEqual( 'Roy');
		component.find('input').simulate('change');
		expect(onChange.mock.calls.length).toBe(1);
	});
});
