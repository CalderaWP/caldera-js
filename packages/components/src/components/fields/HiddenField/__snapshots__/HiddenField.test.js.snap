// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HiddenField  matches snapshot with all props 1`] = `
<div
  className="caldera-field-wrapper caldera-field-wrapper-hidden"
>
  <input
    className="caldera-field caldera-field-hidden"
    id="i11"
    onChange={[Function]}
    type="hidden"
    value="Roy"
  />
</div>
`;
